db/migrate/20220809152005_fix_ibm_power_vc_sti_classes.rb

```diff
@@ -0,0 +1,168 @@
+  end
+
+  class CloudVolumeType < ActiveRecord::Base
+    include ActiveRecord::IdRegions
+    self.inheritance_column = :_type_disabled
+
+    belongs_to :ext_management_system, :foreign_key => :ems_id, :class_name => "FixIbmPowerVcStiClasses::ExtManagementSystem"
+  end
+
+  def up
+    say_with_time("Fixing IBM PowerVC Network STI classes") do
+      network_klass = "ManageIQ::Providers::IbmPowerVc::NetworkManager"
+
+      network_managers = ExtManagementSystem.in_my_region.where(:type => network_klass)
+
+      CloudNetwork.in_my_region
+                  .where(:ext_management_system => network_managers)
+                  .update_all(:type => "#{network_klass}::CloudNetwork")
+      CloudSubnet.in_my_region
+                 .where(:ext_management_system => network_managers)
+                 .update_all(:type => "#{network_klass}::CloudSubnet")
+      FloatingIp.in_my_region
+                .where(:ext_management_system => network_managers)
+                .update_all(:type => "#{network_klass}::FloatingIp")
+      NetworkPort.in_my_region
+                 .where(:ext_management_system => network_managers)
+                 .update_all(:type => "#{network_klass}::NetworkPort")
+      NetworkRouter.in_my_region
+                   .where(:ext_management_system => network_managers)
+                   .update_all(:type => "#{network_klass}::NetworkRouter")
+      SecurityGroup.in_my_region
+                   .where(:ext_management_system => network_managers)
+                   .update_all(:type => "#{network_klass}::SecurityGroup")
+    end
+
+    say_with_time("Fixing IBM PowerVC Storage (Cinder) STI classes") do
+      cinder_klass = "ManageIQ::Providers::IbmPowerVc::StorageManager::CinderManager"
+
+      cinder_managers = ExtManagementSystem.in_my_region.where(:type => cinder_klass)
+
+      CloudVolume.in_my_region
+                 .where(:ext_management_system => cinder_managers)
+                 .update_all(:type => "#{cinder_klass}::CloudVolume")
+      CloudVolumeBackup.in_my_region
+                       .where(:ext_management_system => cinder_managers)
+                       .update_all(:type => "#{cinder_klass}::CloudVolumeBackup")
+      CloudVolumeSnapshot.in_my_region
+                         .where(:ext_management_system => cinder_managers)
+                         .update_all(:type => "#{cinder_klass}::CloudVolumeSnapshot")
+      CloudVolumeType.in_my_region
+                     .where(:ext_management_system => cinder_managers)
+                     .update_all(:type => "#{cinder_klass}::CloudVolumeType")
+    end
+  end
+
+  def down
+    say_with_time("Resetting IBM PowerVC Network STI classes") do
+      network_klass = "ManageIQ::Providers::IbmPowerVc::NetworkManager"
+
+      network_managers = ExtManagementSystem.in_my_region.where(:type => network_klass)
+
+      CloudNetwork.in_my_region
+                  .where(:ext_management_system => network_managers)
+                  .update_all(:type => nil)
```

Comment:
I'm not sure if these are supposed to be reset to `nil` or back to the `ManageIQ::Providers::Openstack::` types. With `nil` would they be reset by the next refresh update?

Reply:
Ideally a down migration would set them to whatever they were before the up migration.